src/communication/stream_transfer_receiver.cpp

```diff
@@ -138,5 +138,9 @@ namespace cubstream
     return m_last_received_position;
   }
 
+  void transfer_receiver::terminate_connection (void)
+  {
+    m_channel.close_connection ();
```

Comment:
Shouldn't `terminate_connection` be synchronized, since from what I understand, m_channel could be accessed from different threads (one is receiver daemon and another one is the called of terminate_connection)

Reply:
m_channel is instance in receiver, it lives until receiver is destroyed.
I don't think we need to protect socket of channel.